environments/pull-environments.sh

```diff
@@ -1,11 +1,79 @@
 #!/bin/bash
 # using bin/bash for macOS and Linux compatibility
-set -e
-IFS='|'
 
 dirs=""
-# Pull the backend for each environment
-for dir in ./*/ ; do
+
+# Get Options from CLI
+
+# Options:
+#  -r (region): AWS region to pull environments from. Defaults to us-east-2.
+#  -i (include): Regex pattern that matches environment folders to pull from.
+#  -e (exclude): Regex pattern that matches environment folders to ignore.
+# If both -i and -e are specified, it'll match all dirs that match -i but not -e.
+
+# Examples:
+# - Include only sms/phone environments:
+#   ./pull-environments.sh -r us-east-2 -i "\./auth/auth-with-((phone-and-sms-mfa)|(phone-number)|(totp-and-sms-mfa))"
+
+while getopts ':r:i:e:' OPTION; do
+
+  case "$OPTION" in
+  r)
+    region="$OPTARG"
+    ;;
+  i)
+    include=$OPTARG
+    ;;
+  e)
+    exclude="$OPTARG"
+    ;;
+
+  ?)
+    echo "Usage: $(basename $0) [-r region] [-i include] [-e exclude]"
+    exit 1
+    ;;
+  esac
+
+done
+
+# Apply defaults
+[ -z "$region" ] && region="us-east-2"        # default to us-east-2
+[ -z "$include" ] && include="\./[a-zA-Z\-]*" # default to all folders in cwd
+
+# Check OS, because `find` implementation is slightly different between OS.
+uname="$(uname)"
+
+# We only support macOS and Linux.
+# TODO: We should rewrite this script in node.js for better cross-os support.
+if [[ "$uname" != "Darwin" && "$uname" != 'Linux' ]]; then
+  echo "ERROR: Unknown os: "$uname""
+  exit 1
+fi
```

Comment:
Here's the part I don't like. Bash scripts are not perfectly compatible, and even supporting both Linux and MacOS (ie. Darwin) is difficult. 

For now, I only tested against those two, which is imo okay because this is only for internal engineers who do not use windows.

Reply:
(even on linux, i assume wsl would work as it's essentially linux, but this is not something we need to worry about yet).